start.py

```diff
@@ -558,40 +563,48 @@ def _command_condition(deployment, command_id, command_description):
+            parcel_version = parcel.version.split('-')[0]
+            if parcel_version == version:
+                logger.info('Detected parcel version matches specified version. Continuing ...')
             else:
                 parcel.deactivate()
+            # Only one parcel can be activated at a time, so once one is found, our work is done.
+            break
 
-                for config in deployment.get_cm_config():
-                    if config['name'] == 'REMOTE_PARCEL_REPO_URLS':
-                        break
-                else:
-                    raise Exception('Failed to find remote parcel repo URLs configuration.')
-                parcel_repo_urls = config['value']
 
-                kafka_parcel_repo_url = '{}/{}'.format(KAFKA_PARCEL_REPO_URL, args.kafka_version)
-                logger.debug('Adding Kafka parcel repo URL (%s) ...', kafka_parcel_repo_url)
-                deployment.update_cm_config(
-                    {'REMOTE_PARCEL_REPO_URLS': '{},{}'.format(parcel_repo_urls,
-                                                               kafka_parcel_repo_url)}
-                )
+def _install_parcel_from_remote_repo(deployment, product, version, parcel_repo_version=None):
+    for config in deployment.get_cm_config():
+        if config['name'] == 'REMOTE_PARCEL_REPO_URLS':
+            break
+    else:
+        raise Exception('Failed to find remote parcel repo URLs configuration.')
+    parcel_repo_urls = config['value']
+
+    service_parcel_repo_url = '{}/{}'.format(SERVICE_PARCEL_REPO_URLS[product],
+                                             parcel_repo_version if parcel_repo_version is not None else version)
+    logger.debug('Adding parcel repo URL (%s) ...', service_parcel_repo_url)
+    deployment.update_cm_config(
+        {'REMOTE_PARCEL_REPO_URLS': '{},{}'.format(parcel_repo_urls,
+                                                   service_parcel_repo_url)}
+    )
 
-                logger.debug('Refreshing parcel repos ...')
-                deployment.refresh_parcel_repos()
-                kafka_parcel = next(parcel
-                                    for parcel in deployment.cluster(DEFAULT_CLUSTER_NAME).parcels
-                                    if parcel.product == 'KAFKA'
-                                    and parcel.version.split('-')[0] == args.kafka_version)
-                kafka_parcel.download().distribute().activate()
-            # Only one parcel can be activated at a time, so once one is found, our work is done.
+    logger.debug('Refreshing parcel repos ...')
+    deployment.refresh_parcel_repos()
+    deployment.cluster(DEFAULT_CLUSTER_NAME).wait_for_parcel_stage(product=product, version=version)
+    service_parcel = deployment.cluster(DEFAULT_CLUSTER_NAME).parcel(product=product, version=version)
+    service_parcel.download().distribute().activate()
+
+
+def _configure_kafka(deployment, cluster, args):
```

Comment:
Let's change the `args` to `kafka_version` (for the same reason as we did it for the other functions).